executor/explainfor_test.go

```diff
@@ -479,7 +479,4 @@ func (s *testPrepareSerialSuite) TestPointGetUserVarPlanCache(c *C) {
 	tk.MustQuery("execute stmt using @a").Check(testkit.Rows(
 		"2 4 2 2",
 	))
-	tk.MustQuery("select @@last_plan_from_cache").Check(testkit.Rows(
-		"1",
-	))
```

Comment:
Sadly, this is the side effect of this PR, we may have false negative decision about "cache it or not", unless we track the whole range calculation procedure to record if any user variable is related to the final ranges. That would be trivial and easy to be broken by new code.

Reply:
Can we change the test results instead of deleting these tests?
Because now we explicitly don't want to cache these plans to assure correctness.